### PR TITLE
CDAP-15851: Repartitioner plugin validation.

### PR DESCRIPTION
**Pull request by AntonGensitskiy**
JIRA: https://issues.cask.co/browse/CDAP-15851
WIKI: https://wiki.cask.co/display/CE/Guidelines+on+Plugin+Validation

In scope of this PR:

- integrated with new validation API

### CODE REVIEW

File: src/main/java/io/cdap/plugin/repartitioner/Repartitioner.java [hunk 1]
```diff
@@ -35,19 +34,17 @@
...
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
     super.configurePipeline(pipelineConfigurer);
-
-    if (config.getPartitions(1) > 0) {
-      throw new IllegalArgumentException("Number of partitions should be greater than zero.");
-    }
+    FailureCollector failureCollector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    config.validate(failureCollector);
```
Comment (mlozbin-cybervisiontech) [inline src/main/java/io/cdap/plugin/repartitioner/Repartitioner.java:47]: missed validation check

File: src/main/java/io/cdap/plugin/repartitioner/RepartitionerConfig.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2017-2019 Cask Data, Inc.
```
Comment (mlozbin-cybervisiontech) [inline src/main/java/io/cdap/plugin/repartitioner/RepartitionerConfig.java]: wrong date

File: src/main/java/io/cdap/plugin/repartitioner/RepartitionerConfig.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  public int getPartitions() {
+    return partitions;
+  }
+
+  public boolean getShuffle() {
+    return shuffle;
+  }
+
+  public void validate(FailureCollector failureCollector) {
+    if (partitions <= 0) {
```
Comment (mlozbin-cybervisiontech) [inline src/main/java/io/cdap/plugin/repartitioner/RepartitionerConfig.java]: what about enabled macro?

File: src/main/java/io/cdap/plugin/repartitioner/RepartitionerConfig.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      failureCollector.addFailure("Number of partitions should be greater than zero.", null)
+        .withConfigProperty(PARTITIONS);
+    }
+  }
+
+  /**
+   * Builder for RepartitionerConfig
+   */
+  public static final class Builder {
```
Comment (mlozbin-cybervisiontech) [inline src/main/java/io/cdap/plugin/repartitioner/RepartitionerConfig.java:79]: I'm not sure we need a builder for 2 fields

File: src/test/java/io/cdap/plugin/repartitioner/RepartitionerConfigTest.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © 2017-2019 Cask Data, Inc.
```
Comment (mlozbin-cybervisiontech) [inline src/test/java/io/cdap/plugin/repartitioner/RepartitionerConfigTest.java]: wrong date

File: pom.xml [hunk 6]
```diff
@@ -28,12 +28,12 @@
```
Comment (mlozbin-cybervisiontech) [inline pom.xml:28]: 1. change url in issueManagement
2. change url in maven-javadoc-plugin

**Review by mlozbin-cybervisiontech — CHANGES_REQUESTED** [review]

File: src/main/java/io/cdap/plugin/repartitioner/Repartitioner.java (hunk 1 above)
Comment (AntonGensitskiy) [reply src/main/java/io/cdap/plugin/repartitioner/Repartitioner.java:47]: The validation check wasn't added based on discussion with Vinisha
https://github.com/data-integrations/topn/pull/12#discussion_r323958674

File: src/main/java/io/cdap/plugin/repartitioner/RepartitionerConfig.java (hunk 2 above)
Comment (AntonGensitskiy) [reply src/main/java/io/cdap/plugin/repartitioner/RepartitionerConfig.java]: Done.

File: src/main/java/io/cdap/plugin/repartitioner/RepartitionerConfig.java (hunk 3 above)
Comment (AntonGensitskiy) [reply src/main/java/io/cdap/plugin/repartitioner/RepartitionerConfig.java]: Added macro check.

File: src/test/java/io/cdap/plugin/repartitioner/RepartitionerConfigTest.java (hunk 5 above)
Comment (AntonGensitskiy) [reply src/test/java/io/cdap/plugin/repartitioner/RepartitionerConfigTest.java]: Done.

File: pom.xml (hunk 6 above)
Comment (AntonGensitskiy) [reply pom.xml:28]: Done.

**Pull request merged**